applications/tube_network/src/statistics.py

```diff
@@ -0,0 +1,234 @@
+        'match',
+        '   $sta isa station, has lat $lat, has name $nam;',
+        '   $lat ' + str(lat) + ';',
+        'get $nam;'
+    ]
+
+    print_to_log("Query:", "\n".join(query))
+    query = "".join(query)
+
+    answers = transaction.query(query).collect_concepts()
+    result = [ answer.value() for answer in answers ]
+
+
+    print_to_log("West most stations with " + str(lat) + " are: ", result)
+
+
+# How long is the longest trip between two stations?
+def execute_query_4(question, transaction):
+    print_to_log("Question: ", question)
+
+    query = 'compute max of duration, in route-section;'
+
+    print_to_log("Query:", query)
+
+    answer = list(transaction.query(query))[0]
+    max_duration = answer.number()
+    print(max_duration)
+
+    query = [
+        'match',
+        '   $rou (section: $sec, origin: $ori, destination: $des, route-operator: $tul) isa route;',
+        '   $sec isa route-section, has duration ' + str(max_duration) + ';',
+        '   $tul isa tube-line, has name $tul-nam;',
+        '   $tun (beginning: $sta1, end: $sta2, service: $sec) isa tunnel;',
+        '   $sta1 isa station, has name $sta1-nam;',
+        '   $sta2 isa station, has name $sta2-nam;',
+        '   $ori isa station, has name $ori-nam;',
+        '   $des isa station, has name $des-nam;',
+        'get;'
+    ]
+
+    print_to_log("Query:", "\n".join(query))
+    query = "".join(query)
+
+    answers = transaction.query(query)
+    for answer in answers:
+        answer = answer.map()
+        print_to_log("Longest trip is found in: ", "Tunnel from " +
+                                                    answer.get("sta1-nam").value() +
+                                                    " to " + answer.get("sta2-nam").value() +
+                                                    ", via " + answer.get("tul-nam").value() +
+                                                    ", on the route going from " +
+                                                    answer.get("ori-nam").value() +
+                                                    " to " + answer.get("des-nam").value())
+
+
+# What's the average duration of all trips?
+def execute_query_5(question, transaction):
+    print_to_log("Question: ", question)
+
+    query = 'compute mean of duration, in route-section;'
+
+    print_to_log("Query:", query)
+
+    answer = list(transaction.query(query))[0]
+    mean_duration = answer.number()
+
+    print("Average duration: " + str(mean_duration))
+
+
+# What's the median duration among all trips?
+def execute_query_6(question, transaction):
+    print_to_log("Question: ", question)
+
+    query = 'compute median of duration, in route-section;'
+
+    print_to_log("Query:", query)
+
+    answer = list(transaction.query(query))[0]
+    median_duration = answer.number()
+
+    print("Median of durations: " + str(median_duration))
+
+
+# What's the median duration among all trips?
+def execute_query_7(question, transaction):
```

Comment:
All of these execute_query functions except 3 and 4 repeat themselves, so they could all be simplified by extracting a method

Reply:
@jmsfltchr we definitely should code clean when we're building products. when it comes to examples though, the objective is to showcase Grakn's features, Graql and usage of Clients. In this case, the query functions show how 1) a question is written in Graql 2) a Graql query gets executed, and 3) answers are retrieved. extracting any part of this code into another method breaks this flow